### PR TITLE
Update OpenJDK 8 to 1.8.232 (from pkgsrc)

### PR DESCRIPTION
**Pull request by Mno-hime**
Release notes: https://www.oracle.com/technetwork/java/javase/8all-relnotes-2226344.html
We miss-out quite some security fixes.

Can't fix this thing:
```
Assembling build/amd64/hotspot/src/os_cpu/solaris_x86/vm/solaris_x86_64.s
build/amd64/hotspot/src/os_cpu/solaris_x86/vm/solaris_x86_64.s: Assembler messages:
build/amd64/hotspot/src/os_cpu/solaris_x86/vm/solaris_x86_64.s:46: Error: bad register name `%rdi)'
build/amd64/hotspot/src/os_cpu/solaris_x86/vm/solaris_x86_64.s:51: Error: bad register name `%rax'
build/amd64/hotspot/src/os_cpu/solaris_x86/vm/solaris_x86_64.s:59: Error: bad register name `%rax'
build/amd64/hotspot/src/os_cpu/solaris_x86/vm/solaris_x86_64.s:72: Error: bad register name `%rdx'
build/amd64/hotspot/src/os_cpu/solaris_x86/vm/solaris_x86_64.s:73: Error: bad register name `%rdx'
build/amd64/hotspot/src/os_cpu/solaris_x86/vm/solaris_x86_64.s:74: Error: bad register name `%rdi'
build/amd64/hotspot/src/os_cpu/solaris_x86/vm/solaris_x86_64.s:75: Error: bad register name `%rdi'
...
build/amd64/hotspot/src/os_cpu/solaris_x86/vm/solaris_x86_64.s:382: Error: bad register name `%rdx'
build/amd64/hotspot/src/os_cpu/solaris_x86/vm/solaris_x86_64.s:384: Error: bad register name `%rdx'
gmake[7]: *** [build/amd64/hotspot/make/solaris/makefiles/rules.make:156: solaris_x86_64.o] Error 1
```

`solaris_x86_64.s` seems to be ready for `gcc -S`, but `gas` is used instead.

### CODE REVIEW

**Comment by despair86** [comment]
>solaris_x86_64.s seems to be ready for gcc -S, but gas is used instead.

because `gcc -S` invokes `/usr/sfw/bin/gas` to do the actual assembly?

File: components/runtime/openjdk-8/patches/patch-jdk_make_lib_ServiceabilityLibraries.gmk.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+Proper libiconv link (and SunOS needs it too).
+
+--- jdk/make/lib/ServiceabilityLibraries.gmk.orig	2018-05-26 23:58:30.000000000 +0000
++++ jdk/make/lib/ServiceabilityLibraries.gmk
+@@ -282,7 +282,7 @@ $(eval $(call SetupNativeCompilation,BUI
+         -framework Cocoa -framework Security -framework ApplicationServices, \
+     LDFLAGS_SUFFIX := $(LIBINSTRUMENT_LDFLAGS_SUFFIX), \
+     LDFLAGS_SUFFIX_macosx := -liconv $(LIBZ), \
+-    LDFLAGS_SUFFIX_solaris := $(LIBZ) -L $(INSTALL_LIBRARIES_HERE)/jli -ljli $(LIBDL) -lc, \
++    LDFLAGS_SUFFIX_solaris := $(LIBICONV) $(LIBZ) -L $(INSTALL_LIBRARIES_HERE)/jli -ljli $(LIBDL) -lc, \
```
Comment (pyhalov) [inline components/runtime/openjdk-8/patches/patch-jdk_make_lib_ServiceabilityLibraries.gmk.patch]: We don't ship GNU libiconv on OI

File: components/runtime/openjdk-8/patches/patch-hotspot_make_solaris_makefiles_gcc.make.patch [hunk 2]
```diff
@@ -28,6 +28,16 @@ GCC support.
  
  ARCHFLAG = $(ARCHFLAG/$(BUILDARCH))
  
+@@ -86,7 +98,8 @@ ARCHFLAG/amd64   = -m64 -march=k8
```
Comment (pyhalov) [inline components/runtime/openjdk-8/patches/patch-hotspot_make_solaris_makefiles_gcc.make.patch]: Build fails due to this change, revert gcc.make.patch

File: components/runtime/openjdk-8/patches/patch-bn.patch [hunk 3]
```diff
@@ -12,28 +12,28 @@ Use pkgsrc fonts.
 -filename.DejaVu_Sans_Bold=/usr/local/lib/X11/fonts/dejavu/DejaVuSans-Bold.ttf
 -filename.DejaVu_Sans_Oblique=/usr/local/lib/X11/fonts/dejavu/DejaVuSans-Oblique.ttf
 -filename.DejaVu_Sans_Bold_Oblique=/usr/local/lib/X11/fonts/dejavu/DejaVuSans-BoldOblique.ttf
-+filename.DejaVu_Sans=/usr/share/fonts/X11/TTF/DejaVuSans.ttf
-+filename.DejaVu_Sans_Bold=/usr/share/fonts/X11/TTF/DejaVuSans-Bold.ttf
-+filename.DejaVu_Sans_Oblique=/usr/share/fonts/X11/TTF/DejaVuSans-Oblique.ttf
-+filename.DejaVu_Sans_Bold_Oblique=/usr/share/fonts/X11/TTF/DejaVuSans-BoldOblique.ttf
++filename.DejaVu_Sans=@PREFIX@/share/fonts/X11/TTF/DejaVuSans.ttf
```
Comment (pyhalov) [inline components/runtime/openjdk-8/patches/patch-bn.patch]: Revert this patch, we don't care about @PREFIX@

File: components/runtime/openjdk-8/patches/patch-common_autoconf_spec.gmk.in.patch [hunk 4]
```diff
@@ -17,7 +17,7 @@ $NetBSD: patch-common_autoconf_spec.gmk.in,v 1.4 2015/06/10 13:29:57 tnn Exp $
  
  # LDFLAGS used to link the jdk native launchers (C-code)
 -LDFLAGS_JDKEXE:=@LDFLAGS_JDKEXE@
-+LDFLAGS_JDKEXE:=-Xlinker -R\$$$$ORIGIN/server @LDFLAGS_JDKEXE@
++LDFLAGS_JDKEXE:=-Xlinker -R@prefix@/java/openjdk8/jre/lib@OPENJDK_TARGET_CPU_LIBDIR@/jli @LDFLAGS_JDKEXE@
```
Comment (pyhalov) [inline components/runtime/openjdk-8/patches/patch-common_autoconf_spec.gmk.in.patch]: I'm not sure, likely this change also should be reverted.

File: components/runtime/openjdk-8/patches/patch-jdk_make_lib_CoreLibraries.gmk.patch [hunk 5]
```diff
@@ -0,0 +1,15 @@
+$NetBSD: patch-jdk_make_lib_CoreLibraries.gmk,v 1.5 2018/02/23 14:43:49 fhajny Exp $
```
Comment (pyhalov) [inline components/runtime/openjdk-8/patches/patch-jdk_make_lib_CoreLibraries.gmk.patch]: This patch is not needed

**Comment by pyhalov** [comment]
@Mno-hime Please, test and review my changes. If everything is fine, feel free to merge the change as one commit.

**Comment by pyhalov** [comment]
Needs work, as illumos-gate  build fails with 
```
dmake: Warning: Target `install' not remade because of errors
Undefined                       first referenced
 symbol                             in file
__stack_chk_fail                    /usr/java/jre/lib/amd64/server/libjvm.so
__stack_chk_guard                   /usr/java/jre/lib/amd64/server/libjvm.so
ld: fatal: symbol referencing errors. No output written to poold
```

**Comment by pyhalov** [comment]
Could build illumos-gate and some simple java programs with this JDK.
@Mno-hime I think we can merge this.

**Pull request merged**